akka-cluster-tools/src/main/scala/akka/cluster/client/ClusterClient.scala

```diff
@@ -470,7 +471,12 @@ object ClusterReceptionistSettings {
 final class ClusterReceptionistSettings(
   val role: Option[String],
   val numberOfContacts: Int,
-  val responseTunnelReceiveTimeout: FiniteDuration) extends NoSerializationVerificationNeeded {
+  val responseTunnelReceiveTimeout: FiniteDuration,
+  val verboseHeartbeatLogging: Boolean) extends NoSerializationVerificationNeeded {
+
+  def this(role: Option[String], numberOfContacts: Int, responseTunnelReceiveTimeout: FiniteDuration) {
```

Comment:
When you add the config property to ClusterSettings I think it's not necessary to mess with the ClusterReceptionistSettings, instead you can use `Cluster(context.system).settings.VerboseHeartbeatLogging directly inside the ClusterReceptionist actor.